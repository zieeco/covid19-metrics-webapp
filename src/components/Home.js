/* eslint-disable react/jsx-no-undef */
import React, { useEffect } from 'react';
import { FaRegArrowAltCircleRight } from 'react-icons/fa';
import { useDispatch, useSelector } from 'react-redux';
import { Link } from 'react-router-dom';
import countryMapSource from '../countryData/countryMaps';
import { fetchCountriesFromAPI } from '../redux/countries/countries';
import BaseCountry from './BaseCountry';
import Form from './Form';

const Home = () => {
  const countries = useSelector((state) => state.countriesReducer);
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(fetchCountriesFromAPI());
  }, []);

  return (
    <section className="countries">
      <BaseCountry />
      <Form />
      <ul className="country-list">
        { countries.length > 0 ? countries.map((country) => {
          const countryMap = countryMapSource(country.name);
          const floatNumber = parseFloat(country.today_confirmed);
          const floatNumberFormat = floatNumber.toLocaleString('en');
          return (
            <li key={country.id} className="country-list-item">
              <Link to={{ pathname: country.name }}>
                <FaRegArrowAltCircleRight className="arrow mb-1" />
                <div className="map-box">
                  <img src={countryMap} alt="country map" className="map-images" />
                </div>
                <p className="country-info fw fs-1">{country.name.toUpperCase()}</p>
                <p className="country-info mb-1 fs-1">{floatNumberFormat}</p>
              </Link>
            </li>
          );
        }) : ''}
      </ul>
    </section>
  );
};

export default Home;
