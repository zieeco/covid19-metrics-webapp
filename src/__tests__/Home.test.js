import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import store from '../redux/configureStore';
import Home from '../components/Home';

const HomeStoreProvider = () => (
  <Provider store={store}>
    <Home />
  </Provider>
);

describe('test for Home component', () => {
  it('renders correctly', () => {
    const RenderHomeComponent = renderer.create(<HomeStoreProvider />).toJSON();
    expect(RenderHomeComponent).toMatchSnapshot();
  });
});
