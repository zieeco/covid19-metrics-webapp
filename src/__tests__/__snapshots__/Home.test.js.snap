// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test for Home component renders correctly 1`] = `
<section
  className="countries"
>
  <div
    className="base-container flex"
  />
  <div
    className="form-sect"
  >
    <div
      className="form-container flex"
    >
      <p
        className="fw"
      >
        STATS BY COUNTRY
      </p>
      <form
        action="#"
        className="flex"
      >
        <input
          onChange={[Function]}
          placeholder="search by country name..."
          type="text"
        />
      </form>
    </div>
  </div>
  <ul
    className="country-list"
  >
    <div
      className="not-found flex"
    >
      <div
        className="error-box flex"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
              "height": "40px",
              "width": "40px",
            }
          }
          viewBox="0 0 576 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M569.517 440.013C587.975 472.007 564.806 512 527.94 512H48.054c-36.937 0-59.999-40.055-41.577-71.987L246.423 23.985c18.467-32.009 64.72-31.951 83.154 0l239.94 416.028zM288 354c-25.405 0-46 20.595-46 46s20.595 46 46 46 46-20.595 46-46-20.595-46-46-46zm-43.673-165.346l7.418 136c.347 6.364 5.609 11.346 11.982 11.346h48.546c6.373 0 11.635-4.982 11.982-11.346l7.418-136c.375-6.874-5.098-12.654-11.982-12.654h-63.383c-6.884 0-12.356 5.78-11.981 12.654z"
          />
        </svg>
        <p>
          No Country Found!
        </p>
      </div>
    </div>
  </ul>
</section>
`;
